src/atlas_assistant/tools/sql.py

```diff
@@ -49,15 +106,26 @@ def generate_sql(query: str, runtime: ToolRuntime[Context, State]) -> Command[No
                     tool_call_id=runtime.tool_call_id,
                 )
             ],
-            "sql_query": sql_query.query,
+            "sql_query": sql_query,
         }
     )
 
 
-def get_prompt(query: str) -> str:
+def get_prompt(dataset: Dataset) -> str:
     return f"""I want you to act like a data scientist.
 
-You will generate an SQL query that will be used with DuckDB to answer a query.
+You will generate between three and six things:
+
+   - An SQL select statement
+   - AN SQL where statement
+   - An optional SQL group by statement
+   - An optional SQL order by statement
+   - An optional SQL limit statement
+   - A brief explanation of why you chose what you did
+
+The SQL should be valid DuckDB SQL.
+
+The dataset schema that the SQL will be used against is:
 
-The query is: {query}
+    {dataset.get_schema_table()}
```

Comment:
I would inject the head as well, and maybe even listing unique values in categorical columns. That should help the model to make better choices.